webapi/Skills/ChatSkills/SemanticChatMemoryExtractor.cs

```diff
@@ -165,12 +174,11 @@ internal static class SemanticChatMemoryExtractor
                 );
             }
         }
-        catch (Exception connectorException)
+        catch (Exception connectorException) when (!connectorException.IsCriticalException())
```

Comment:
Do we need try/catch this here? The exception should already be thrown when it's searched above, right?

If not, I would suggest moving this into the try catch block above as well to reduce duped code.

```
if ((memories?.Count ?? 0) == 0)
            {
                await semanticTextMemory.SaveInformationAsync(
                    collection: memoryCollectionName,
@@ -165,12 +174,11 @@ internal static class SemanticChatMemoryExtractor
                );
            }
```

Reply:
[Connector](https://github.com/microsoft/chat-copilot/pull/365)
https://github.com/microsoft/chat-copilot/issues/351
https://github.com/microsoft/chat-copilot/issues/215

Reply:
Yes, but since `semanticTextMemory.SearchAsync` is already called above, the missing index exception should already be thrown on that call

I also think you can consolidate both calls into the same try catch block 
![image](https://github.com/microsoft/chat-copilot/assets/125500434/6a03d14f-b38c-4de1-b847-3e60bbc2c957)

Reply:
Never mind, I see the first one can throw a 404 if there's no related memories